charts/redpanda/templates/_helpers.tpl

```diff
@@ -582,6 +585,97 @@ return a warning if the chart is configured with insufficient CPU
   {{- toJson $brokers -}}
 {{- end -}}
 
+{{- define "kafka-brokers-sasl-enabled" -}}
+  {{- $root := . -}}
+  {{- $kafkaService := .Values.listeners.kafka }}
+  {{- $auditLogging := .Values.auditLogging }}
+  {{- $brokers := list -}}
+  {{- $broker_tls := dict -}}
+  {{- $result := dict -}}
+  {{- $tlsEnabled := .Values.tls.enabled -}}
+  {{- $tlsCerts := .Values.tls.certs -}}
+  {{- $trustStoreFile := "" -}}
+  {{- $requireClientAuth := dig "tls" "requireClientAuth" false $kafkaService -}}
+  {{- if and ( eq "internal" $auditLogging.listener ) ( eq (default "sasl" $kafkaService.authenticationMethod) "sasl" ) -}}
+    {{- range $address := (include "seed-server-list" $root | mustFromJson) }}
+      {{- $brokerItem := ( dict
+        "address" $address
+        "port" $kafkaService.port
+        )
+      -}}
+      {{- $brokers = append $brokers $brokerItem -}}
+    {{- end -}}
+    {{- if $brokers -}}
+      {{- $result = set $result "brokers" $brokers -}}
+    {{- end -}}
+    {{- if dig "tls" "enabled" $tlsEnabled $kafkaService -}}
+      {{- $cert := get $tlsCerts $kafkaService.tls.cert -}}
+      {{- if empty $cert -}}
+        {{- fail (printf "Certificate, '%s', used but not defined") -}}
+      {{- end -}}
+      {{- if $cert.caEnabled -}}
+        {{- $trustStoreFile =  ( printf "/etc/tls/certs/%s/ca.crt" $kafkaService.tls.cert ) -}}
+      {{- else -}}
+        {{- $trustStoreFile = "/etc/ssl/certs/ca-certificates.crt" -}}
+      {{- end -}}
+      {{- $broker_tls = ( dict
+        "enabled" true
+        "cert_file" ( printf "/etc/tls/certs/%s/tls.crt" $kafkaService.tls.cert )
+        "key_file" ( printf "/etc/tls/certs/%s/tls.key" $kafkaService.tls.cert )
+        "require_client_auth" $requireClientAuth
+        )
+      -}}
+      {{- if $trustStoreFile -}}
+        {{- $broker_tls = set $broker_tls "truststore_file" $trustStoreFile -}}
+      {{- end -}}
+      {{- if $broker_tls -}}
+        {{- $result = set $result "broker_tls" $broker_tls -}}
+      {{- end -}}
+    {{- end -}}
+  {{- else -}}
+    {{- range $name, $listener := $kafkaService.external -}}
+      {{- if and $listener.port $name (dig "enabled" true $listener) ( eq (default "sasl" $listener.authenticationMethod) "sasl" ) ( eq $name $auditLogging.listener ) -}}
```

Comment:
```suggestion
      {{- if and $listener.port $name (dig "enabled" false $listener) ( eq (default "sasl" $listener.authenticationMethod) "sasl" ) ( eq $name $auditLogging.listener ) -}}
```

I don't understand why the default is `true`?

Reply:
because it is. It is usually commented out and assumed to be enabled. The override exists in case you wish to enable it, or disable it on purpose.